lib/multi_precision.cairo

```diff
@@ -16,128 +16,315 @@ func mod_range{range_check_ptr}(dividend : felt) -> (remainder : felt):
     return (remainder)
 end
 
-# @dev truncates overflow on most signifigant bit
-func multi_precision_add{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(
-    x : BigInt6, y : BigInt6
-) -> (res : BigInt6):
-    alloc_locals
+# Public namespace for multi_precision math
+namespace multi_precision:
```

Comment:
It should be easier to see what was done by looking at the file rather than the diff. 

1. function names now omit the `multi_precision_` part
2. internal functions are moved outside of the namespace. (Can be persuaded to keep them inside the namespace)